demo/src/main/java/org/gbif/pipelines/demo/DwCA2InterpretedRecordsPipeline.java

```diff
@@ -0,0 +1,248 @@
+import org.gbif.pipelines.io.avro.ExtendedRecord;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.beam.runners.direct.DirectRunner;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.coders.AvroCoder;
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.AvroIO;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.join.CoGbkResult;
+import org.apache.beam.sdk.transforms.join.CoGroupByKey;
+import org.apache.beam.sdk.transforms.join.KeyedPCollectionTuple;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.beam.sdk.values.TupleTag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.gbif.pipelines.core.functions.transforms.RawToInterpretedCategoryTransformer.spatialCategory;
+import static org.gbif.pipelines.core.functions.transforms.RawToInterpretedCategoryTransformer.spatialCategoryIssues;
+import static org.gbif.pipelines.core.functions.transforms.RawToInterpretedCategoryTransformer.temporalCategory;
+import static org.gbif.pipelines.core.functions.transforms.RawToInterpretedCategoryTransformer.temporalCategoryIssues;
+
+/**
+ * A simple demonstration showing a pipeline running locally which will read UntypedOccurrence from a DwC-A file
+ * transform it into interpreted occurence records
+ * .
+ */
+public class DwCA2InterpretedRecordsPipeline {
+
+  private static final TupleTag<Event> temporalTag = new TupleTag<Event>() {};
+  private static final TupleTag<Location> spatialTag = new TupleTag<Location>() {};
+  private static final TupleTag<IssueLineageRecord> temporalIssueTag = new TupleTag<IssueLineageRecord>() {};
+  private static final TupleTag<IssueLineageRecord> spatialIssueTag = new TupleTag<IssueLineageRecord>() {};
+  private static final Logger LOG = LoggerFactory.getLogger(DwCA2InterpretedRecordsPipeline.class);
+
+  public static void main(String[] args) {
+    PipelineOptions options = PipelineOptionsFactory.create();
+    options.setRunner(DirectRunner.class); // forced for this demo
+    Pipeline p = Pipeline.create(options);
+
+    // register Avro coders for serializing our messages
+    Coders.registerAvroCoders(p,
+                              ExtendedRecord.class,
+                              Event.class,
+                              Location.class,
+                              ExtendedOccurence.class,
+                              Issue.class,
+                              Lineage.class,
+                              IssueLineageRecord.class);
+    p.getCoderRegistry()
```

Comment:
move all this to a registerCoders method? just to hide this 'nice' (Beam!!) way of doing things